### PR TITLE
Remove grpcio from minimal dependency

### PR DESCRIPTION
**Pull request by rynewang**
Removes grpcio from `ray` dependency while adds it to `ray[default]` dependency.

Removed use of grpc client: `gcs_node_info_stub.DrainNode` in autoscaler. Moved it into `raylet.pyx`.
Removed use of grpc client: `gcs_node_resources_stub.GetAllResourceUsage` in autoscaler. Moved it into `raylet.pyx`.
Removed use of grpc status code. Moved it into `raylet.pyx`.
Removed Ray Client tests in `test_runtime_env_ray_minimal.py`

### CODE REVIEW

**Review by edoakes — APPROVED** [review]
Stamp requirements change

File: python/ray/autoscaler/_private/autoscaler.py
Comment (pcmoritz) [inline python/ray/autoscaler/_private/autoscaler.py:693]: I believe `logger.exception` prints the backtrace, so this might print it twice, which would be undesirable :)

**Review by pcmoritz — APPROVED** [review]

File: python/ray/autoscaler/_private/autoscaler.py
Comment (rynewang) [reply python/ray/autoscaler/_private/autoscaler.py]: done.

File: python/ray/tests/test_usage_stats.py
Comment (rkooo567) [inline python/ray/tests/test_usage_stats.py:128]: Add this to the test definition? 

```
@pytest.skipif(...)
def test_get_extra_usage_tags_to_report(...)
```

File: python/ray/tests/test_basic.py
Comment (rkooo567) [inline python/ray/tests/test_basic.py:1119]: can you add `assert os.environ.get("RAY_MINIMAL") == 1` here?

File: python/ray/tests/conftest.py
Comment (rkooo567) [inline python/ray/tests/conftest.py:653]: if we add this for individual tests, there are a lot of them? Also, is it possible to add this to the def like the comment below>? 

```
@pytest.skipif(...)
def start_cluster(..)
```

File: python/ray/autoscaler/_private/autoscaler.py
Comment (rkooo567) [inline python/ray/autoscaler/_private/autoscaler.py:690]: No need to handle `UNIMPLEMENTED` here?

**Review by rkooo567 — CHANGES_REQUESTED** [review]
Before merging, let's actually make sure we have right error messages the client is not supported by minimal ray. 

I think 2 places need to be changed;

1. At the top of this doc, we should mention it is supported only when ray[default] is installed.
https://docs.ray.io/en/latest/cluster/running-applications/job-submission/ray-client.html
2. When users trying to connect to ray client via `ray://` protocol, we should raise an exception if it is minimal ray.

Also If it is minimal ray, what's happening to client server? is it not even started now?

**Comment by pcmoritz** [comment]
> Before merging, let's actually make sure we have right error messages the client is not supported by minimal ray.
> 
> I think 2 places need to be changed;
> 
> 1. At the top of this doc, we should mention it is supported only when ray[default] is installed.
>    https://docs.ray.io/en/latest/cluster/running-applications/job-submission/ray-client.html
> 2. When users trying to connect to ray client via `ray://` protocol, we should raise an exception if it is minimal ray.
> 
> Also If it is minimal ray, what's happening to client server? is it not even started now?

+1 to the error message -- but let's please say `ray[client]` needs to be installed. We also want to remove grpcio from `ray[default]` going forward and `ray[client]` should be the only thing depending on it in the future :)

File: python/ray/tests/test_usage_stats.py
Comment (rynewang) [reply python/ray/tests/test_usage_stats.py:128]: how do we conditionally skip the test with `skipif`? I tried various ways and this approach is the most understandable one to me.

File: python/ray/tests/test_basic.py
Comment (rynewang) [reply python/ray/tests/test_basic.py:1119]: I don't see the point. It's possible someone has grpc installed on their own, and installed minimal ray, and run this test.

File: python/ray/autoscaler/_private/autoscaler.py
Comment (rynewang) [reply python/ray/autoscaler/_private/autoscaler.py:690]: The document mentioned it used to deal with Ray version < 1.8.0, and now it's 2.7.0 so I guess we would not ever handle the UNIMPLEMENTED case?

**Review by pcmoritz — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Failed tests seem unrelated

**Comment by rkooo567** [comment]
(Also, let's make sure to mention this to the release note. Please track the changes so that  we can record them!)

**Comment by rkooo567** [comment]
Looks like we need a doc approval. cc @angelinalg can you give us an approval?

**Review by pcmoritz — APPROVED** [review]
doc approval

**Pull request merged**
